custom/icds_reports/data_pull/queries.py

```diff
@@ -0,0 +1,59 @@
+from django.db import connections
+from django.utils.functional import cached_property
+
+
+class BaseQuery:
+    name = ""
+    query_file_path = ""
+
+    def run(self, db_alias):
+        with open(self.result_file_name, "w") as output:
+            db_conn = connections[db_alias]
+            c = db_conn.cursor()
+            query = self.sql_query
+            query = query.replace('\n', ' ')
+            c.copy_expert(
+                "COPY ({query}) TO STDOUT DELIMITER ',' CSV HEADER;".format(query=query),
+                output)
+        return self.result_file_name
+
+    @property
+    def result_file_name(self):
+        return "%s.csv" % self.name
+
+    @cached_property
+    def raw_query(self):
+        filepath = 'custom/icds_reports/data_pull/sql_queries/%s.sql' % self.query_file_path
+        with open(filepath) as _sql:
+            sql = _sql.read()
+        return sql
+
+    def sql_query(self):
+        raise NotImplemented
+
+
+class MonthBasedQuery(BaseQuery):
+    def __init__(self, month):
+        self.month = month
+
+    @property
+    def result_file_name(self):
+        return "%s-%s.csv" % (self.name, self.month)
+
+    @cached_property
+    def sql_query(self):
+        return self.raw_query.format(month=self.month)
```

Comment:
is there not like a parameterized sql way to do this rather than use python string formatting?

Reply:
May be, would that be better?

Reply:
I think so as it would future proof against potential SQL injections

Reply:
Looked a bit but could not find a way to do that here.
For now the value from HQ is coming from the form and is cleaned so we should be good here. But definitely open to suggestions to look further.